examples/experimental/dagster-airlift/dagster_airlift/in_airflow/dagster_run_utils.py

```diff
@@ -0,0 +1,70 @@
+from typing import Any, Mapping, NamedTuple, Optional
+
+TERMINAL_STATI = ["SUCCESS", "FAILURE", "CANCELED"]
+SYSTEM_TAG_PREFIX = "dagster/"
+MAX_RETRIES_TAG = f"{SYSTEM_TAG_PREFIX}max_retries"
+RETRY_NUMBER_TAG = f"{SYSTEM_TAG_PREFIX}retry_number"
+PARENT_RUN_ID_TAG = f"{SYSTEM_TAG_PREFIX}parent_run_id"
+SUCCESS_STATUS = "SUCCESS"
+RETRY_ON_ASSET_OR_OP_FAILURE_TAG = f"{SYSTEM_TAG_PREFIX}retry_on_asset_or_op_failure"
+RUN_FAILURE_REASON_TAG = f"{SYSTEM_TAG_PREFIX}failure_reason"
+STEP_FAILURE_REASON = "STEP_FAILURE"
+
+
+class DagsterRunResult(NamedTuple):
+    status: str
+    tags: Mapping[str, Any]
+
+    @property
+    def run_retries_configured(self) -> bool:
+        return MAX_RETRIES_TAG in self.tags
+
+    @property
+    def has_remaining_retries(self) -> bool:
+        if MAX_RETRIES_TAG not in self.tags:
+            raise Exception(
+                "Tried to retrieve tags from run, but run retries "
+                f"were either not set or not properly configured. Found tags: {self.tags}"
+            )
+        return self.max_retries - self.retry_number > 0
+
+    @property
+    def run_will_automatically_retry(self) -> bool:
+        if not self.run_retries_configured:
+            return False
+        if (
+            not self.should_retry_on_asset_or_op_failure
+            and self.failure_reason == STEP_FAILURE_REASON
+        ):
+            return False
+        return self.has_remaining_retries
+
+    @property
+    def should_retry_on_asset_or_op_failure(self) -> bool:
+        return get_boolean_tag_value(self.tags.get(RETRY_ON_ASSET_OR_OP_FAILURE_TAG), True)
```

Comment:
this can be(and most often is) a deployment/instance setting rather than a tag on individual runs, so I don't think this is going to work in all cases

Reply:
The tag can be used to override the value for individual runs

I wonder if we need the run monitoring machinery to stamp the run with a tag or something so that it's easy for callsites to know if the run is going to retry or not

Reply:
Ah - I assumed that the tag would also be set on the run if it's not overridden explicitly.